library/src/main/java/com/google/android/exoplayer/hls/HlsPlaylistParser.java

```diff
@@ -216,13 +213,15 @@ private static HlsMediaPlaylist parseMediaPlaylist(LineIterator iterator, Uri ba
       } else if (line.equals(DISCONTINUITY_TAG)) {
         segmentDiscontinuity = true;
       } else if (!line.startsWith("#")) {
+        String thisSegmentEncryptionIV =
```

Comment:
Looking more closely at this, presumably you should only generate the segmentMediaSequence derived encryptionIV if the content is actually encrypted as AES-128. In which case you need an extra condition. Something like:

```
String thisSegmentEncryptionIV;
if (segmentEncryptionIV != null) {
  thisSegmentEncryptionIV = segmentEncryptionIV;
} else if (HlsMediaPlaylist.ENCRYPTION_METHOD_AES_128.equals(segmentEncryptionMethod)) {
  thisSegmentEncryptionIV = Integer.toHexString(segmentMediaSequence);
} else {
  thisSegmentEncryptionIV = null;
}
```

Does that make sense? Sorry, should have seen that the first time around!